test/mocha/40-database.js

```diff
@@ -0,0 +1,298 @@
+        const {executionStats} = await brZcapStorage.zcaps.get({
+          controller,
+          referenceId,
+          explain: true
+        });
+        executionStats.nReturned.should.equal(1);
+        executionStats.totalKeysExamined.should.equal(1);
+        executionStats.totalDocsExamined.should.equal(1);
+        executionStats.executionStages.inputStage.inputStage.inputStage.stage
+          .should.equal('IXSCAN');
+        executionStats.executionStages.inputStage.inputStage.inputStage
+          .keyPattern.should.eql({controller: 1, referenceId: 1});
+      });
+    it(`is properly indexed for 'controller' and 'id' in get()`,
+      async () => {
+        const {capability, controller} = mockData.zcaps.alpha;
+        const {executionStats} = await brZcapStorage.zcaps.get({
+          controller,
+          id: capability.id,
+          explain: true
+        });
+        executionStats.nReturned.should.equal(1);
+        executionStats.totalKeysExamined.should.equal(1);
+        executionStats.totalDocsExamined.should.equal(1);
+        executionStats.executionStages.inputStage.inputStage.inputStage.stage
+          .should.equal('IXSCAN');
+        // winning query plan is {controller: 1, referenceId: 1} in this case.
+        executionStats.executionStages.inputStage.inputStage.inputStage
+          .keyPattern.should.eql({controller: 1, referenceId: 1});
+      });
+    it(`is properly indexed for 'controller' and 'referenceId' in find()`,
+      async () => {
+        const {controller, referenceId} = mockData.zcaps.alpha;
+        const query = {
+          controller: helpers.hash(controller),
+          referenceId: helpers.hash(referenceId)
+        };
+        const {executionStats} = await brZcapStorage.zcaps.find({
+          query,
+          explain: true
+        });
+        executionStats.nReturned.should.equal(1);
+        executionStats.totalKeysExamined.should.equal(1);
+        executionStats.totalDocsExamined.should.equal(1);
+        executionStats.executionStages.inputStage.stage.should.equal('IXSCAN');
+        executionStats.executionStages.inputStage.keyPattern.should.eql({
+          controller: 1, referenceId: 1});
+      });
+    it(`is properly indexed for 'controller' and 'id' in find()`,
+      async () => {
+        const {capability, controller} = mockData.zcaps.alpha;
+        const query = {
+          controller: helpers.hash(controller),
+          id: helpers.hash(capability.id)
+        };
+        const {executionStats} = await brZcapStorage.zcaps.find({
+          query,
+          explain: true
+        });
+        executionStats.nReturned.should.equal(1);
+        executionStats.totalKeysExamined.should.equal(1);
+        executionStats.totalDocsExamined.should.equal(1);
+        executionStats.executionStages.inputStage.stage.should.equal('IXSCAN');
+        // winning query plan is {controller: 1, referenceId: 1} in this case.
```

Comment:
Is this always going to be true? We discussed this before -- can we add more mock data to force it to use the expected index? I would imagine that this might have a different result on different machines as well. Can we allow for `{controller: 1, id: 1}` to be selected here as well -- if we can't force it with more mock data (preferred)?